usecases/traverser/hybrid_group_by.go

```diff
@@ -27,33 +28,71 @@ func (e *Explorer) groupSearchResults(ctx context.Context, sr search.Results, gr
 	for _, result := range sr {
 		prop_i := result.Object().Properties
 		prop := prop_i.(map[string]interface{})
-		val, ok := prop[groupBy.Property].(string)
+		rawValue := prop[groupBy.Property]
 
-		if !ok {
-			continue
+		values, err := extractGroupByValues(rawValue)
+		if err != nil {
+			return nil, fmt.Errorf("failed to extract groupBy values for property %s: %w", groupBy.Property, err)
 		}
 
-		current, groupExists := groups[val]
-		if len(current) >= groupBy.ObjectsPerGroup {
+		if len(values) == 0 {
 			continue
 		}
 
-		if !groupExists && len(groups) >= groupBy.Groups {
-			continue
-		}
+		// Process all values for this result, but stop if we hit the groups limit
+		skipResult := false
+		for _, val := range values {
+			if skipResult {
+				break
+			}
+
+			current, groupExists := groups[val]
+			if len(current) >= groupBy.ObjectsPerGroup {
+				continue
+			}
+
+			if !groupExists && len(groups) >= groupBy.Groups {
+				skipResult = true
+				break
+			}
 
-		groups[val] = append(current, result)
+			groups[val] = append(current, result)
 
-		if !groupExists {
-			// this group doesn't exist add it to the ordered list
-			groupsOrdered = append(groupsOrdered, val)
+			if !groupExists {
+				// this group doesn't exist add it to the ordered list
+				groupsOrdered = append(groupsOrdered, val)
+			}
 		}
 	}
 
 	out := make(search.Results, 0, len(sr))
 	for i, groupValue := range groupsOrdered {
 		groupMembers := groups[groupValue]
-		first := groupMembers[0]
+
+		// Create a copy of the first result to avoid modifying the original
+		original := groupMembers[0]
+		first := search.Result{
+			ID:                 original.ID,
+			DocID:              original.DocID,
+			ClassName:          original.ClassName,
+			Score:              original.Score,
+			SecondarySortValue: original.SecondarySortValue,
+			ExplainScore:       original.ExplainScore,
+			Dist:               original.Dist,
+			Vector:             original.Vector,
+			Vectors:            original.Vectors,
+			Beacon:             original.Beacon,
+			Certainty:          original.Certainty,
+			Schema:             original.Schema,
+			Created:            original.Created,
+			Updated:            original.Updated,
+			VectorWeights:      original.VectorWeights,
+			IsConsistent:       original.IsConsistent,
+			Tenant:             original.Tenant,
+			Dims:               original.Dims,
+			// Create a new AdditionalProperties map instead of sharing
+			AdditionalProperties: models.AdditionalProperties{},
+		}
```

Comment:
Can we revert this change back to just `first := groupMembers[0]`, reasoning behind this is that now with this change would need to always remember to manually update this code whenever we add or remove need field in `search.Result`